tests/CMakeLists.txt

```diff
@@ -5,11 +5,14 @@
 # Arguments:
 #   TARGET_NAME       : name of the test (required)
 #   SOURCE_FILE       : source file, defaults to <TARGET_NAME>.cpp (optional)
-#   KOKKOS_TOOLS_LIBS : the test environment will received the variable 'KOKKOS_TOOLS_LIBS' that is set as the path
+#   KOKKOS_TOOLS_LIBS : the test environment will receive the variable 'KOKKOS_TOOLS_LIBS' that is set as the path
 #                       to the target file of this argument (optional)
-function(kp_add_executable_and_test)
+#   KOKKOS_TOOLS_SAMPLER_VERBOSE : the test environment will receive the variable 'KOKKOS_TOOLS_SAMPLER_VERBOSE' that is set as the value of 1 for printing the sample has been taken 
+#   KOKKOS_TOOLS_GLOBALFENCES : test environment receives the variable 'KOKKOS_TOOLS_GLOBALFENCES' that is set as the value of 1 to turn the tool's auto-fencing on. 
+#   KOKKOS_TOOLS_SAMPLER_SKIP : test environment receives the variable 'KOKKOS_TOOLS_SAMPLER_SKIP' that is set as the value of the number of Kokkos kernel invocations to skip before a tooling activity is invoked. 
 
-    cmake_parse_arguments(kaeat_args "" "TARGET_NAME;SOURCE_FILE;KOKKOS_TOOLS_LIBS" "" ${ARGN})
+function(kp_add_executable_and_test)
+    cmake_parse_arguments(kaeat_args "" "TARGET_NAME;SOURCE_FILE;KOKKOS_TOOLS_SAMPLER_VERBOSE;KOKKOS_TOOLS_GLOBALFENCES;KOKKOS_TOOLS_SAMPLER_SKIP" "KOKKOS_TOOLS_LIBS" ${ARGN})
```

Comment:
Please explain what changed here.

Reply:
These are the environment variables needed for the sampler test. One needs to add these in order to control parameters of the sampler utility. Also, the other demangling test doesn't use global fences. For the sampler's test, we need to use global fences to test the sampler.

Reply:
Why do we need global fences here? In the end, we only care about the strings printed and not if the kernels actually ran, right? It would be different if the kernels themselves were printing stuff but all kernels are empty anyway.

Reply:
To follow-up, see #117 where fences in the sampler are used in the same way I am doing. 

In terms why we need to test with fences enabled in the sampler, the kernels themselves (more specifically the kernel callbacks themselves) are printing because of the kernel-logger.

Reply:
I have taken out the setting of global fences as you requested.

(Also, I was getting confused on if you were saying to get rid of global fences in sampling altogether).